app/controllers/gift_requests_controller.rb

```diff
@@ -36,7 +38,18 @@ def destroy_multiple
   end
 
   private
-    def gift_request_params
-      params.require(:gift_request).permit(:id, :name, :recipient, :description, :link, :gender, :age)
+
+  def gift_request_params
+    params.require(:gift_request).permit(:id, :name, :recipient, :description, :link, :gender, :age)
+  end
+
+  def require_owned_gift_request!
+    if params[:action] == 'edit'
+      gift_id = params[:gift_request_id]
+    else
+      gift_id = params[:id]
     end
+
+    current_user.role.lists.map{|list| list.gift_requests.map{|gr| gr.id}}.flatten.include?(gift_id)
```

Comment:
if you want to do something like `things.map{|thing| thing.other_thing}` you can just do `things.map(&:other_thing)` instead.

more to the point, i'd suggest adding `has_many :gift_requests, through: :lists` to Therapist so you can just do `current_user.role.gift_requests.pluck(:id).include?(gift_id)`